### DISCUSSION TITLE
How do I get the Mocha example to work

### DISCUSSION THREAD

**Discussion by theGhostJW** [pact-foundation/pact-js#845] [Q&A] [answered]
Hi,
I have copied the Mocha example and updated it for Typescript in this [repo](https://github.com/theGhostJW/pact-play/tree/mocha).

When I run `npm` or `yarn test` (Windows 10) I get the following:
```
  2 failing

  1) Pact between MyConsumer and MyProvider
       with 30000 ms timeout for Pact
         get /dogs
           returns the correct response:
     Error: connect ECONNREFUSED ::1:8992
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1158:16)

  2) Pact between MyConsumer and MyProvider
       with 30000 ms timeout for Pact
         "after each" hook: MochaPactVerifyAfterEach for "returns the correct response":
     Error: Pact verification failed - expected interactions did not match actual.
      at new VerificationError (node_modules\@pact-foundation\pact\src\errors\verificationError.js:19:42)
      at C:\pact-play\node_modules\@pact-foundation\src\httpPact.ts:157:17
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
```
Looks like I am not getting a connection to the mock server. I have also tried just coping the directory without modification and running `npm install` then `npm test`. This gives me the same result.

What else do I need to do to make this work?  

Thanks in advance to anyone who can help.

Initially posted on [stackoverflow](https://stackoverflow.com/questions/71568098/how-do-i-get-a-basic-example-of-a-pact-test-working-with-typescript-and-mocha)

**Comment by mefellows** [comment, marked as answer]
It looks like it could be an ipv6 issue. Could you please try binding explicitly to an address (not a hostname).

e.g. 127.0.0.1

**Reply by theGhostJW** [reply]
yep that fixed it.
Thanks @mefellows !!!

**Reply by mefellows** [reply]
ha, good to know! Thanks for accepting, I'll add to the troubleshooting guide.

**Comment by mefellows** [comment]
I just ran your repository locally and got this

```


  Pact between MyConsumer and MyProvider
    with 30000 ms timeout for Pact
      get /dogs
        ✔ returns the correct response
      get /dog/1
        ✔ returns the correct response


  2 passing (7s)

➜  pact-play git:(main) ✗
```


Could you please set log level to debug and share the output?

**Reply by mefellows** [reply]
I see you're on windows, so it could be the long path issue (search issues for that). Trying moving project closer to `c:\`
